trl/trainer/dpo_trainer.py

```diff
@@ -1627,6 +1633,72 @@ def concatenated_forward(
 
         return output
 
+    def _compute_multi_loss(
+        self,
+        chosen_logps: torch.FloatTensor,
+        rejected_logps: torch.FloatTensor,
+        ref_chosen_logps: torch.FloatTensor,
+        ref_rejected_logps: torch.FloatTensor,
+        model_output: dict = None,
+    ) -> tuple[torch.FloatTensor, torch.FloatTensor, torch.FloatTensor]:
+        """
+        Compute combined loss for multiple loss types (e.g., "sigmoid,bco_pair,sft" for MPO).
+
+        This enables combining different loss formulations with individual weights.
+        For example, MPO uses DPO (sigmoid) as preference loss, BCO (bco_pair) as quality loss,
+        and SFT (sft) as generation loss.
+        """
+        # Store original loss type
+        original_loss_type = self.loss_type
+        loss_type_list = original_loss_type.split(",")
```

Comment:
I think we should also add
```
loss_type_list = list(map(lambda x: x.strip(), loss_type_list))
```

The rationale behind it is if a user adds spaces between comma-separated loss functions, the script will not error out.